datatypes/src/operations/reproject.rs

```diff
@@ -218,8 +218,111 @@ where
     }
 }
 
+/// This method calculates a suggested pixel size for the translation of a raster into a different projection.
+/// The source raster is described using a `BoundingBox2D` and a pixel size as `SpatialResolution`.
+/// A suggested pixel size is calculated using the approach used by GDAL:
+/// The upper left and the lower right coordinates of the bounding box are projected in the target SRS.
+/// Then, the distance between both points in the target SRS is devided by the distance in pixels of the source.
+pub fn suggest_pixel_size_like_gdal<P: CoordinateProjection>(
+    bbox: BoundingBox2D,
+    spatial_resolution: SpatialResolution,
+    projector: &P,
+) -> Result<SpatialResolution> {
+    // calculate the number of pixels per axis
+    let x_pixels = (bbox.size_x() / spatial_resolution.x).abs();
+    let y_pixels = (bbox.size_y() / spatial_resolution.y).abs();
+    // get the diagonal distance between bbox edges
+    let diag_pixels: f64 = (x_pixels * x_pixels + y_pixels * y_pixels).sqrt();
+
+    // reproject the upper left and lower right coordinates to the target srs.
+    // NOTE: the edges of the projected bbox might differ.
+    let proj_ul_coord = bbox.upper_left().reproject(projector)?;
+    let proj_lr_coord = bbox.lower_right().reproject(projector)?;
+
+    // calculate the distance between upper left and lower right coordinate in srs units
+    let proj_ul_lr_vector = proj_ul_coord - proj_lr_coord;
+    let proj_ul_lr_distance = (proj_ul_lr_vector.x * proj_ul_lr_vector.x
+        + proj_ul_lr_vector.y * proj_ul_lr_vector.y)
+        .sqrt();
+
+    // derive the pixel size by deviding srs unit distance by pixel distance in the source bbox
+    let proj_ul_lr_pixel_size = proj_ul_lr_distance / diag_pixels;
```

Comment:
Is it possible that the bounding box is just one coordinate and is diag_pixels then zero? This would panic, I guess we need a test for that.

Reply:
i never thought about this. I guess we have to do some research

Reply:
i think this must not happen. if the bbox is really for a single pixel, then it must be pixel lower left to pixel upper right and therefore it is not allowed to be zero. However, i add a test for this.

Reply:
4316575